File: rb/features.rb
# BrewpageHosting SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module BrewpageHostingFeatures
  def self.make_feature(name)
    case name
    when "base"
      BrewpageHostingBaseFeature.new
    when "test"
      BrewpageHostingTestFeature.new
    else
      BrewpageHostingBaseFeature.new
    end
  end
end

File: py/brewpagehosting_sdk/features.py
# BrewpageHosting SDK feature factory

from brewpagehosting_sdk.feature.base_feature import BrewpageHostingBaseFeature
from brewpagehosting_sdk.feature.test_feature import BrewpageHostingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BrewpageHostingBaseFeature(),
        "test": lambda: BrewpageHostingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
